components/relevancy/src/lib.rs

```diff
@@ -47,9 +57,53 @@ impl RelevancyStore {
     ///
     ///  This method may execute for a long time and should only be called from a worker thread.
     #[handle_error(Error)]
-    pub fn ingest(&self, _top_urls_by_frecency: Vec<String>) -> ApiResult<()> {
+    pub fn ingest(&self, top_urls_by_frecency: Vec<String>) -> ApiResult<HashMap<Interest, u32>> {
         populate_interests::ensure_interest_data_populated(&self.db)?;
-        todo!()
+        let mut interest_counter: HashMap<Interest, u32> = HashMap::new();
+
+        let hashed_top_urls: Vec<UrlHash> = top_urls_by_frecency
+            .into_iter()
+            .filter_map(|url| hash_url(&url))
+            .collect();
+
+        let classified_urls = RelevancyStore::get_url_interest_data();
+        for classified_url in classified_urls {
+            if let Some(url) = &classified_url.url {
+                match hashed_top_urls.contains(url) {
+                    true => {
+                        *interest_counter.entry(classified_url.interest).or_insert(0) += 1;
+                    }
+                    false => {
+                        *interest_counter.entry(Interest::Inconclusive).or_insert(0) += 1;
+                    }
+                }
+            } else {
+                *interest_counter.entry(Interest::Inconclusive).or_insert(0) += 1;
+            }
+        }
+        Ok(interest_counter)
+    }
+
+    /// Temp method until we actually retrieve data from RS
+    pub fn get_url_interest_data() -> Vec<URLInterest> {
```

Comment:
Using a temporary method for this makes a lot of sense to me.

I'm my comment above was right, I think this should return a HashMap.  Alternatively, maybe this could insert rows into a database.  I think using a db would be closer to how things will work in the final product and there are some methods in `db.rs` that should work.  Maybe it's better to do that in a follow-up PR though.  WDYT?

Reply:
Agreed, we'd definitely want to do more things around the db for the final product instead of loading everything into memory for each classification.

For interest data ingestion, I think we'd periodically ingest the interest data from Remote Settings and persist it to the db, we will figure out a reasonable schema to facilitate the subsequent interest classifications.

For classification, we will query against the db for each URL hash to get its associated interest category if any. Note that as we've decided to store raw URL hashes (as opposed to other compact structures e.g. Bloom filters ) for each interest on Remote Settings, each input URL/hash would require a point lookups against the db, which should be fine for small/medium size inputs. There are ways to reduce the overhead for large inputs and we can look into those in the future.

Does that align with what's on your mind? Would love to hear your thoughts.

Reply:
Yup, this is what I was thinking as well.